### DISCUSSION TITLE
how to get ecInstanceId on click 3d obejct?

### DISCUSSION THREAD

**Discussion by Andromeda-Main-Line** [iTwin/itwinjs-core#3896] [Q&A] [answered]
I got into itwin.js for a week.
Finished tutorial and get used to this framework.

I've got another sample .dgn file and implemented in my test project.
I saw unstructured than tutorials sample.
Too hard to find target 3d object's ecInstanceId.

So, I planned to find target ecInstanceId from iTwinjs View first, instead of searching it from iModelDB.
(I saw someone did it, youtube)

Look into API doc, but I couldn't find how to set simple click event on the iTwinjs 3d view.
If it is possible, I guess there is ecInstanceId in the callback return parameter.

Could anyone know how to do it?

**Comment by pmconne** [comment, marked as answer]
It will be easier to help you find the right solution if you state your high-level goal ("I want my application to allow the user to...") rather than specific technical hurdles encountered when attempting a particular solution.

It sounds like you might be trying to [write a Tool](https://www.itwinjs.org/learning/frontend/tools/) that does something to elements when the user clicks on them.
Or perhaps you just want to [be notified](https://www.itwinjs.org/reference/core-frontend/selectionset/selectionsetevent/) when the contents of the iModel's [selection set](https://www.itwinjs.org/reference/core-frontend/selectionset/selectionset/) changes?

**Reply by Andromeda-Main-Line** [reply]
Thank you for your generous attention.

I'm trying to get ecInstanceId when the user clicks on 3d object.

Hard to find, how to set click event trigger on twin.js view.

**Reply by pmconne** [reply]
You're welcome. Did you read my response, or the information I linked you to? There is no "set click event trigger on twin.js view" that will give you an ECInstanceId. Please explain your goal ***from a user perspective*** so we can assist you.

**Reply by Andromeda-Main-Line** [reply]
Oh, I see. I just read your link and it was helpful :D (I didn't think it was link)
I was too hard to get specific 3d object's ecInstanceId bia SQL,
So I decide to get id with event callback.
I should chang my method of approch.
Have a nice day :D
